src/RegisterContext.js

```diff
@@ -211,8 +211,7 @@ RegisterContext.prototype = {
     options = options || {};
 
     if(!this.registered && !options.all) {
-      this.logger.warn('already unregistered');
-      return;
```

Comment:
Hmmmm... it is not immediately clear to me how removing this `return` breaks all of our tests.